### ISSUE TITLE
Auto-generated merge commit is causing the GitHub Action failing

### ISSUE THREAD

**Issue by SiegeSailor** [opensource-nepal/commitlint#78]
**Describe the bug**

I resolved a merge conflict by committing on the GitHub Web UI, which creates a commit with the following message:

```
Merge branch 'main' into release
    
Signed-off-by: Jin Yu Zhang <39463416+SiegeSailor@users.noreply.github.com>
```

However, the GitHub Action `commitlint` reports _Commit message does not follow the Conventional Commits format_ for such a commit message: 

- `pull_request` with the `synchronize` event type:
  - `commitlint` pulls out all the commit messages and therefore reports the error message every time
  -  see https://github.com/SiegeSailor/Smarty-Notebook/actions/runs/19212728519/job/54917412427
- `push`
  - `commitlint` only validates the current commit message, so only the action triggered by the merge commit reports the error message
  - See https://github.com/SiegeSailor/Smarty-Notebook/actions/runs/19205599978/job/54900288644 for the failed case
  - See https://github.com/SiegeSailor/Smarty-Notebook/actions/runs/19212728100/job/54917411463 for the successful case

**To Reproduce**

Create a merge commit on the GitHub UI.

**Expected behavior**

`commitlint` ignores the auto-generated merge commits.

**Environment**

-   Operating system: `ubuntu-latest`

**Additional Context**

`commitlint` (1.3.1) doesn't report any error when I run it locally:

```shell
commitlint "Merge branch 'main' into release\nSigned-off-by: Jin Yu Zhang <39463416+SiegeSailor@users.noreply.github.com>"
Commit validation: successful!
```

```shell
commitlint "Merge branch 'main' into release"
Commit validation: successful!
```
